File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let already=[words[0]]
    let cnt=0;
    for(let i=1;i<words.length;i++){
        if(words[i-1].slice(words[i-1].length-1)!==words[i].slice(0,1)||already.includes(words[i])){
            return (i+1)%n===0? [n,(i+1)/n]: [(i+1)%n,Math.floor((i+1)/n)+1]
        }
        else already.push(words[i])
    }
    return [0, 0]

    //return answer;
}

File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(int n, vector<string> words) {
    vector<int> answer;
    vector<string> ald;
    
    ald.push_back(words[0]);
    for(int i=1;i<words.size();i++){
        auto it = find(ald.begin(),ald.end(),words[i]);
        if(words[i-1][words[i-1].size()-1]!=words[i][0]||
           it!=ald.end()){
            if((i+1)%n==0){
                answer.push_back(n);
                answer.push_back((i+1)/n);
                return answer;
            }else{
                answer.push_back((i+1)%n);
                answer.push_back((i+1)/n+1);
                return answer;
            }
        }
        ald.push_back(words[i]);
        
    }
     answer.push_back(0);
    answer.push_back(0);
    
    return answer;
}